app/classifier/drawing-tools/freehand-segment-shape.jsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DrawingToolRoot from './root';
+import DeleteButton from './delete-button';
+import { createPathFromCoords, filterDupeCoords, roundCoords } from './freehand-helpers';
+
+const BUFFER = 16;
+const DELETE_BUTTON_WIDTH = 8;
+const FINISHER_RADIUS = 8;
+const POINT_RADIUS = 4;
+
+class FreehandSegmentShapeTool extends React.Component {
+  static initCoords = null;
+
+  static defaultValues() {
+    return {
+      points: [],
+      _inProgress: false,
+      _currentlyDrawing: false
+    };
+  }
+
+  static initStart(coords, mark) {
+    mark.points.push(roundCoords(coords));
+    mark._inProgress = true;
+    mark._currentlyDrawing = true;
+    return mark;
+  }
+
+  static initMove(coords, mark) {
+    mark.points.push(roundCoords(coords));
+    return mark;
+  }
```

Comment:
Same comment here about returning a new object instead of mutating the mark directly.